File: website/src/lib/locations.ts
export interface CountryData {
  name: string;
  states: StateData[];
}

export interface StateData {
  name: string;
  cities: string[];
}

export const countries: CountryData[] = [
  {
    name: "Nigeria",
    states: [
      { name: "Abia", cities: ["Aba", "Umuahia", "Arochukwu"] },
      { name: "Adamawa", cities: ["Yola", "Mubi", "Jimeta"] },
      { name: "Akwa Ibom", cities: ["Uyo", "Ikot Ekpene", "Eket"] },
      { name: "Anambra", cities: ["Awka", "Onitsha", "Nnewi"] },
      { name: "Bauchi", cities: ["Bauchi", "Azare", "Jama'are"] },
      { name: "Bayelsa", cities: ["Yenagoa", "Brass", "Sagbama"] },
      { name: "Benue", cities: ["Makurdi", "Gboko", "Otukpo"] },
      { name: "Borno", cities: ["Maiduguri", "Biu", "Bama"] },
      { name: "Cross River", cities: ["Calabar", "Ikom", "Ogoja"] },
      { name: "Delta", cities: ["Asaba", "Warri", "Sapele"] },
      { name: "Ebonyi", cities: ["Abakaliki", "Afikpo", "Onueke"] },
      { name: "Edo", cities: ["Benin City", "Auchi", "Ekpoma"] },
      { name: "Ekiti", cities: ["Ado Ekiti", "Ikere", "Ikole"] },
      { name: "Enugu", cities: ["Enugu", "Nsukka", "Abakaliki"] },
      { name: "FCT", cities: ["Abuja", "Gwagwalada", "Kuje"] },
      { name: "Gombe", cities: ["Gombe", "Kaltungo", "Billiri"] },
      { name: "Imo", cities: ["Owerri", "Orlu", "Okigwe"] },
      { name: "Jigawa", cities: ["Dutse", "Hadejia", "Kazaure"] },
      { name: "Kaduna", cities: ["Kaduna", "Zaria", "Kafanchan"] },
      { name: "Kano", cities: ["Kano", "Wudil", "Bichi"] },
      { name: "Katsina", cities: ["Katsina", "Daura", "Funtua"] },
      { name: "Kebbi", cities: ["Birnin Kebbi", "Argungu", "Yauri"] },
      { name: "Kogi", cities: ["Lokoja", "Okene", "Kabba"] },
      { name: "Kwara", cities: ["Ilorin", "Offa", "Jebba"] },
      { name: "Lagos", cities: ["Lagos", "Ikeja", "Lekki", "Surulere", "Epe"] },
      { name: "Nasarawa", cities: ["Lafia", "Keffi", "Akwanga"] },
      { name: "Niger", cities: ["Minna", "Bida", "Suleja"] },
      { name: "Ogun", cities: ["Abeokuta", "Sagamu", "Ijebu Ode"] },
      { name: "Ondo", cities: ["Akure", "Ondo", "Owo"] },
      { name: "Osun", cities: ["Osogbo", "Ile-Ife", "Ilesa"] },
      { name: "Oyo", cities: ["Ibadan", "Oyo", "Ogbomoso"] },
      { name: "Plateau", cities: ["Jos", "Pankshin", "Shendam"] },
      { name: "Rivers", cities: ["Port Harcourt", "Bonny", "Eleme"] },
      { name: "Sokoto", cities: ["Sokoto", "Gusau", "Wurno"] },
      { name: "Taraba", cities: ["Jalingo", "Wukari", "Serti"] },
      { name: "Yobe", cities: ["Damaturu", "Potiskum", "Nguru"] },
      { name: "Zamfara", cities: ["Gusau", "Kaura Namoda", "Anka"] },
    ],
  },
  {
    name: "Ghana",
    states: [
      { name: "Greater Accra", cities: ["Accra", "Tema", "Madina"] },
      { name: "Ashanti", cities: ["Kumasi", "Obuasi", "Ejisu"] },
      { name: "Western", cities: ["Takoradi", "Sekondi", "Axim"] },
      { name: "Central", cities: ["Cape Coast", "Winneba", "Kasoa"] },
      { name: "Eastern", cities: ["Koforidua", "Nkawkaw", "Akosombo"] },
      { name: "Volta", cities: ["Ho", "Keta", "Hohoe"] },
      { name: "Northern", cities: ["Tamale", "Yendi", "Savelugu"] },
      { name: "Upper East", cities: ["Bolgatanga", "Navrongo", "Bawku"] },
      { name: "Upper West", cities: ["Wa", "Jirapa", "Nandom"] },
      { name: "Bono", cities: ["Sunyani", "Berekum", "Dormaa Ahenkro"] },
      { name: "Bono East", cities: ["Techiman", "Kintampo", "Nkoranza"] },
      { name: "Ahafo", cities: ["Goaso", "Bechem", "Kenyasi"] },
      { name: "Oti", cities: ["Dambai", "Kete Krachi", "Nkwanta"] },
      { name: "Western North", cities: ["Sefwi Wiawso", "Bibiani", "Enchi"] },
      { name: "North East", cities: ["Nalerigu", "Walewale", "Gambaga"] },
      { name: "Savannah", cities: ["Damongo", "Salaga", "Bole"] },
    ],
  },
  {
    name: "Kenya",
    states: [
      { name: "Nairobi", cities: ["Nairobi", "Westlands", "Embakasi"] },
      { name: "Mombasa", cities: ["Mombasa", "Nyali", "Likoni"] },
      { name: "Kisumu", cities: ["Kisumu", "Ahero", "Maseno"] },
      { name: "Nakuru", cities: ["Nakuru", "Naivasha", "Gilgil"] },
      { name: "Eldoret", cities: ["Eldoret", "Iten", "Kapsabet"] },
      { name: "Kiambu", cities: ["Kiambu", "Thika", "Ruiru"] },
      { name: "Machakos", cities: ["Machakos", "Athi River", "Kangundo"] },
      { name: "Kakamega", cities: ["Kakamega", "Bungoma", "Mumias"] },
      { name: "Nyeri", cities: ["Nyeri", "Karatina", "Othaya"] },
      { name: "Meru", cities: ["Meru", "Chuka", "Maua"] },
      { name: "Kilifi", cities: ["Kilifi", "Malindi", "Watamu"] },
      { name: "Narok", cities: ["Narok", "Kilgoris", "Lemek"] },
      { name: "Garissa", cities: ["Garissa", "Dadaab", "Hola"] },
      { name: "Kisii", cities: ["Kisii", "Ogembo", "Suneka"] },
      { name: "Murang'a", cities: ["Murang'a", "Kangema", "Kiharu"] },
      { name: "Bungoma", cities: ["Bungoma", "Webuye", "Kimilili"] },
    ],
  },
  {
    name: "South Africa",
    states: [
      { name: "Gauteng", cities: ["Johannesburg", "Pretoria", "Soweto", "Sandton"] },
      { name: "Western Cape", cities: ["Cape Town", "Stellenbosch", "Paarl"] },
      { name: "KwaZulu-Natal", cities: ["Durban", "Pietermaritzburg", "Newcastle"] },
      { name: "Eastern Cape", cities: ["Gqeberha", "East London", "Mthatha"] },
      { name: "Free State", cities: ["Bloemfontein", "Welkom", "Sasolburg"] },
      { name: "Mpumalanga", cities: ["Mbombela", "Witbank", "Secunda"] },
      { name: "Limpopo", cities: ["Polokwane", "Tzaneen", "Phalaborwa"] },
      { name: "North West", cities: ["Mahikeng", "Rustenburg", "Klerksdorp"] },
      { name: "Northern Cape", cities: ["Kimberley", "Upington", "Springbok"] },
    ],
  },
  {
    name: "Uganda",
    states: [
      { name: "Kampala", cities: ["Kampala", "Makindye", "Kawempe"] },
      { name: "Wakiso", cities: ["Wakiso", "Entebbe", "Nansana"] },
      { name: "Mukono", cities: ["Mukono", "Seeta", "Katosi"] },
      { name: "Jinja", cities: ["Jinja", "Njeru", "Kakira"] },
      { name: "Mbarara", cities: ["Mbarara", "Kyazanga", "Bushenyi"] },
      { name: "Gulu", cities: ["Gulu", "Lacor", "Layibi"] },
      { name: "Mbale", cities: ["Mbale", "Bugema", "Lwakhakha"] },
      { name: "Masaka", cities: ["Masaka", "Bukomansimbi", "Kalungu"] },
      { name: "Arua", cities: ["Arua", "Pakwach", "Nebbi"] },
      { name: "Fort Portal", cities: ["Fort Portal", "Kasese", "Bundibugyo"] },
    ],
  },
  {
    name: "Tanzania",
    states: [
      { name: "Dar es Salaam", cities: ["Dar es Salaam", "Kigamboni", "Temeke"] },
      { name: "Dodoma", cities: ["Dodoma", "Kondoa", "Mpwapwa"] },
      { name: "Mwanza", cities: ["Mwanza", "Musoma", "Geita"] },
      { name: "Arusha", cities: ["Arusha", "Moshi", "Karatu"] },
      { name: "Tanga", cities: ["Tanga", "Muheza", "Pangani"] },
      { name: "Mbeya", cities: ["Mbeya", "Tukuyu", "Mbozi"] },
      { name: "Morogoro", cities: ["Morogoro", "Kilosa", "Ifakara"] },
      { name: "Zanzibar", cities: ["Zanzibar City", "Stone Town", "Nungwi"] },
    ],
  },
  {
    name: "Cameroon",
    states: [
      { name: "Centre", cities: ["Yaoundé", "Mbalmayo", "Obala"] },
      { name: "Littoral", cities: ["Douala", "Edéa", "Nkongsamba"] },
      { name: "Southwest", cities: ["Buea", "Limbe", "Kumba"] },
      { name: "Northwest", cities: ["Bamenda", "Kumbo", "Wum"] },
      { name: "West", cities: ["Bafoussam", "Dschang", "Foumban"] },
      { name: "South", cities: ["Ebolowa", "Kribi", "Sangmélima"] },
      { name: "Adamawa", cities: ["Ngaoundéré", "Meiganga", "Tignère"] },
      { name: "North", cities: ["Garoua", "Guider", "Pitoa"] },
      { name: "Far North", cities: ["Maroua", "Kousseri", "Mokolo"] },
    ],
  },
  {
    name: "Senegal",
    states: [
      { name: "Dakar", cities: ["Dakar", "Pikine", "Guédiawaye"] },
      { name: "Thiès", cities: ["Thiès", "Mbour", "Tivaouane"] },
      { name: "Saint-Louis", cities: ["Saint-Louis", "Richard-Toll", "Dagana"] },
      { name: "Diourbel", cities: ["Diourbel", "Mbacké", "Bambey"] },
      { name: "Kaolack", cities: ["Kaolack", "Nioro du Rip", "Guinguinéo"] },
      { name: "Ziguinchor", cities: ["Ziguinchor", "Bignona", "Oussouye"] },
      { name: "Louga", cities: ["Louga", "Linguère", "Kébémer"] },
      { name: "Fatick", cities: ["Fatick", "Sokone", "Foundiougne"] },
      { name: "Tambacounda", cities: ["Tambacounda", "Bakel", "Kidira"] },
      { name: "Kolda", cities: ["Kolda", "Vélingara", "Saraya"] },
    ],
  },
  {
    name: "United States",
    states: [
      { name: "California", cities: ["Los Angeles", "San Francisco", "San Diego", "Sacramento"] },
      { name: "New York", cities: ["New York City", "Buffalo", "Rochester", "Albany"] },
      { name: "Texas", cities: ["Houston", "Dallas", "Austin", "San Antonio"] },
      { name: "Florida", cities: ["Miami", "Orlando", "Tampa", "Jacksonville"] },
      { name: "Illinois", cities: ["Chicago", "Springfield", "Peoria"] },
      { name: "Georgia", cities: ["Atlanta", "Savannah", "Augusta"] },
      { name: "Maryland", cities: ["Baltimore", "Annapolis", "Silver Spring"] },
      { name: "New Jersey", cities: ["Newark", "Jersey City", "Trenton"] },
      { name: "Virginia", cities: ["Richmond", "Virginia Beach", "Arlington"] },
      { name: "Massachusetts", cities: ["Boston", "Cambridge", "Worcester"] },
    ],
  },
  {
    name: "United Kingdom",
    states: [
      { name: "England", cities: ["London", "Manchester", "Birmingham", "Liverpool", "Leeds"] },
      { name: "Scotland", cities: ["Edinburgh", "Glasgow", "Aberdeen", "Dundee"] },
      { name: "Wales", cities: ["Cardiff", "Swansea", "Newport"] },
      { name: "Northern Ireland", cities: ["Belfast", "Derry", "Lisburn"] },
    ],
  },
  {
    name: "Canada",
    states: [
      { name: "Ontario", cities: ["Toronto", "Ottawa", "Mississauga", "Hamilton"] },
      { name: "Quebec", cities: ["Montreal", "Quebec City", "Laval", "Gatineau"] },
      { name: "British Columbia", cities: ["Vancouver", "Victoria", "Surrey", "Burnaby"] },
      { name: "Alberta", cities: ["Calgary", "Edmonton", "Red Deer"] },
      { name: "Manitoba", cities: ["Winnipeg", "Brandon", "Steinbach"] },
      { name: "Saskatchewan", cities: ["Saskatoon", "Regina", "Prince Albert"] },
      { name: "Nova Scotia", cities: ["Halifax", "Sydney", "Dartmouth"] },
    ],
  },
  {
    name: "Other",
    states: [
      { name: "Other", cities: ["Other"] },
    ],
  },
];

export function getStates(countryName: string): StateData[] {
  const country = countries.find((c) => c.name === countryName);
  return country ? country.states : [];
}

export function getCities(countryName: string, stateName: string): string[] {
  const country = countries.find((c) => c.name === countryName);
  if (!country) return [];
  const state = country.states.find((s) => s.name === stateName);
  return state ? state.cities : [];
}

export const referralSources = [
  "Google Search",
  "TikTok",
  "WhatsApp Status",
  "Xlounge-Extreme",
  "From a friend or Family member",
  "Facebook",
  "Myngul",
  "Twitter",
  "Instagram",
  "Youtube",
  "Telegram",
  "Other, please specify",
];

File: lib/core/data/locations.dart
class CountryData {
  final String name;
  final List<StateData> states;
  const CountryData({required this.name, required this.states});
}

class StateData {
  final String name;
  final List<String> cities;
  const StateData({required this.name, required this.cities});
}

const List<CountryData> countries = [
  CountryData(name: 'Nigeria', states: [
    StateData(name: 'Abia', cities: ['Aba', 'Umuahia', 'Arochukwu']),
    StateData(name: 'Adamawa', cities: ['Yola', 'Mubi', 'Jimeta']),
    StateData(name: 'Akwa Ibom', cities: ['Uyo', 'Ikot Ekpene', 'Eket']),
    StateData(name: 'Anambra', cities: ['Awka', 'Onitsha', 'Nnewi']),
    StateData(name: 'Bauchi', cities: ['Bauchi', 'Azare', "Jama'are"]),
    StateData(name: 'Bayelsa', cities: ['Yenagoa', 'Brass', 'Sagbama']),
    StateData(name: 'Benue', cities: ['Makurdi', 'Gboko', 'Otukpo']),
    StateData(name: 'Borno', cities: ['Maiduguri', 'Biu', 'Bama']),
    StateData(name: 'Cross River', cities: ['Calabar', 'Ikom', 'Ogoja']),
    StateData(name: 'Delta', cities: ['Asaba', 'Warri', 'Sapele']),
    StateData(name: 'Ebonyi', cities: ['Abakaliki', 'Afikpo', 'Onueke']),
    StateData(name: 'Edo', cities: ['Benin City', 'Auchi', 'Ekpoma']),
    StateData(name: 'Ekiti', cities: ['Ado Ekiti', 'Ikere', 'Ikole']),
    StateData(name: 'Enugu', cities: ['Enugu', 'Nsukka', 'Abakaliki']),
    StateData(name: 'FCT', cities: ['Abuja', 'Gwagwalada', 'Kuje']),
    StateData(name: 'Gombe', cities: ['Gombe', 'Kaltungo', 'Billiri']),
    StateData(name: 'Imo', cities: ['Owerri', 'Orlu', 'Okigwe']),
    StateData(name: 'Jigawa', cities: ['Dutse', 'Hadejia', 'Kazaure']),
    StateData(name: 'Kaduna', cities: ['Kaduna', 'Zaria', 'Kafanchan']),
    StateData(name: 'Kano', cities: ['Kano', 'Wudil', 'Bichi']),
    StateData(name: 'Katsina', cities: ['Katsina', 'Daura', 'Funtua']),
    StateData(name: 'Kebbi', cities: ['Birnin Kebbi', 'Argungu', 'Yauri']),
    StateData(name: 'Kogi', cities: ['Lokoja', 'Okene', 'Kabba']),
    StateData(name: 'Kwara', cities: ['Ilorin', 'Offa', 'Jebba']),
    StateData(name: 'Lagos', cities: ['Lagos', 'Ikeja', 'Lekki', 'Surulere', 'Epe']),
    StateData(name: 'Nasarawa', cities: ['Lafia', 'Keffi', 'Akwanga']),
    StateData(name: 'Niger', cities: ['Minna', 'Bida', 'Suleja']),
    StateData(name: 'Ogun', cities: ['Abeokuta', 'Sagamu', 'Ijebu Ode']),
    StateData(name: 'Ondo', cities: ['Akure', 'Ondo', 'Owo']),
    StateData(name: 'Osun', cities: ['Osogbo', 'Ile-Ife', 'Ilesa']),
    StateData(name: 'Oyo', cities: ['Ibadan', 'Oyo', 'Ogbomoso']),
    StateData(name: 'Plateau', cities: ['Jos', 'Pankshin', 'Shendam']),
    StateData(name: 'Rivers', cities: ['Port Harcourt', 'Bonny', 'Eleme']),
    StateData(name: 'Sokoto', cities: ['Sokoto', 'Gusau', 'Wurno']),
    StateData(name: 'Taraba', cities: ['Jalingo', 'Wukari', 'Serti']),
    StateData(name: 'Yobe', cities: ['Damaturu', 'Potiskum', 'Nguru']),
    StateData(name: 'Zamfara', cities: ['Gusau', 'Kaura Namoda', 'Anka']),
  ]),
  CountryData(name: 'Ghana', states: [
    StateData(name: 'Greater Accra', cities: ['Accra', 'Tema', 'Madina']),
    StateData(name: 'Ashanti', cities: ['Kumasi', 'Obuasi', 'Ejisu']),
    StateData(name: 'Western', cities: ['Takoradi', 'Sekondi', 'Axim']),
    StateData(name: 'Central', cities: ['Cape Coast', 'Winneba', 'Kasoa']),
    StateData(name: 'Eastern', cities: ['Koforidua', 'Nkawkaw', 'Akosombo']),
    StateData(name: 'Volta', cities: ['Ho', 'Keta', 'Hohoe']),
    StateData(name: 'Northern', cities: ['Tamale', 'Yendi', 'Savelugu']),
    StateData(name: 'Upper East', cities: ['Bolgatanga', 'Navrongo', 'Bawku']),
    StateData(name: 'Upper West', cities: ['Wa', 'Jirapa', 'Nandom']),
    StateData(name: 'Bono', cities: ['Sunyani', 'Berekum', 'Dormaa Ahenkro']),
    StateData(name: 'Bono East', cities: ['Techiman', 'Kintampo', 'Nkoranza']),
    StateData(name: 'Ahafo', cities: ['Goaso', 'Bechem', 'Kenyasi']),
    StateData(name: 'Oti', cities: ['Dambai', 'Kete Krachi', 'Nkwanta']),
    StateData(name: 'Western North', cities: ['Sefwi Wiawso', 'Bibiani', 'Enchi']),
    StateData(name: 'North East', cities: ['Nalerigu', 'Walewale', 'Gambaga']),
    StateData(name: 'Savannah', cities: ['Damongo', 'Salaga', 'Bole']),
  ]),
  CountryData(name: 'Kenya', states: [
    StateData(name: 'Nairobi', cities: ['Nairobi', 'Westlands', 'Embakasi']),
    StateData(name: 'Mombasa', cities: ['Mombasa', 'Nyali', 'Likoni']),
    StateData(name: 'Kisumu', cities: ['Kisumu', 'Ahero', 'Maseno']),
    StateData(name: 'Nakuru', cities: ['Nakuru', 'Naivasha', 'Gilgil']),
    StateData(name: 'Eldoret', cities: ['Eldoret', 'Iten', 'Kapsabet']),
    StateData(name: 'Kiambu', cities: ['Kiambu', 'Thika', 'Ruiru']),
    StateData(name: 'Machakos', cities: ['Machakos', 'Athi River', 'Kangundo']),
    StateData(name: 'Kakamega', cities: ['Kakamega', 'Bungoma', 'Mumias']),
    StateData(name: 'Nyeri', cities: ['Nyeri', 'Karatina', 'Othaya']),
    StateData(name: 'Meru', cities: ['Meru', 'Chuka', 'Maua']),
    StateData(name: 'Kilifi', cities: ['Kilifi', 'Malindi', 'Watamu']),
    StateData(name: 'Narok', cities: ['Narok', 'Kilgoris', 'Lemek']),
    StateData(name: 'Garissa', cities: ['Garissa', 'Dadaab', 'Hola']),
    StateData(name: 'Kisii', cities: ['Kisii', 'Ogembo', 'Suneka']),
    StateData(name: "Murang'a", cities: ["Murang'a", 'Kangema', 'Kiharu']),
    StateData(name: 'Bungoma', cities: ['Bungoma', 'Webuye', 'Kimilili']),
  ]),
  CountryData(name: 'South Africa', states: [
    StateData(name: 'Gauteng', cities: ['Johannesburg', 'Pretoria', 'Soweto', 'Sandton']),
    StateData(name: 'Western Cape', cities: ['Cape Town', 'Stellenbosch', 'Paarl']),
    StateData(name: 'KwaZulu-Natal', cities: ['Durban', 'Pietermaritzburg', 'Newcastle']),
    StateData(name: 'Eastern Cape', cities: ['Gqeberha', 'East London', 'Mthatha']),
    StateData(name: 'Free State', cities: ['Bloemfontein', 'Welkom', 'Sasolburg']),
    StateData(name: 'Mpumalanga', cities: ['Mbombela', 'Witbank', 'Secunda']),
    StateData(name: 'Limpopo', cities: ['Polokwane', 'Tzaneen', 'Phalaborwa']),
    StateData(name: 'North West', cities: ['Mahikeng', 'Rustenburg', 'Klerksdorp']),
    StateData(name: 'Northern Cape', cities: ['Kimberley', 'Upington', 'Springbok']),
  ]),
  CountryData(name: 'Uganda', states: [
    StateData(name: 'Kampala', cities: ['Kampala', 'Makindye', 'Kawempe']),
    StateData(name: 'Wakiso', cities: ['Wakiso', 'Entebbe', 'Nansana']),
    StateData(name: 'Mukono', cities: ['Mukono', 'Seeta', 'Katosi']),
    StateData(name: 'Jinja', cities: ['Jinja', 'Njeru', 'Kakira']),
    StateData(name: 'Mbarara', cities: ['Mbarara', 'Kyazanga', 'Bushenyi']),
    StateData(name: 'Gulu', cities: ['Gulu', 'Lacor', 'Layibi']),
    StateData(name: 'Mbale', cities: ['Mbale', 'Bugema', 'Lwakhakha']),
    StateData(name: 'Masaka', cities: ['Masaka', 'Bukomansimbi', 'Kalungu']),
    StateData(name: 'Arua', cities: ['Arua', 'Pakwach', 'Nebbi']),
    StateData(name: 'Fort Portal', cities: ['Fort Portal', 'Kasese', 'Bundibugyo']),
  ]),
  CountryData(name: 'Tanzania', states: [
    StateData(name: 'Dar es Salaam', cities: ['Dar es Salaam', 'Kigamboni', 'Temeke']),
    StateData(name: 'Dodoma', cities: ['Dodoma', 'Kondoa', 'Mpwapwa']),
    StateData(name: 'Mwanza', cities: ['Mwanza', 'Musoma', 'Geita']),
    StateData(name: 'Arusha', cities: ['Arusha', 'Moshi', 'Karatu']),
    StateData(name: 'Tanga', cities: ['Tanga', 'Muheza', 'Pangani']),
    StateData(name: 'Mbeya', cities: ['Mbeya', 'Tukuyu', 'Mbozi']),
    StateData(name: 'Morogoro', cities: ['Morogoro', 'Kilosa', 'Ifakara']),
    StateData(name: 'Zanzibar', cities: ['Zanzibar City', 'Stone Town', 'Nungwi']),
  ]),
  CountryData(name: 'Cameroon', states: [
    StateData(name: 'Centre', cities: ['Yaoundé', 'Mbalmayo', 'Obala']),
    StateData(name: 'Littoral', cities: ['Douala', 'Edéa', 'Nkongsamba']),
    StateData(name: 'Southwest', cities: ['Buea', 'Limbe', 'Kumba']),
    StateData(name: 'Northwest', cities: ['Bamenda', 'Kumbo', 'Wum']),
    StateData(name: 'West', cities: ['Bafoussam', 'Dschang', 'Foumban']),
    StateData(name: 'South', cities: ['Ebolowa', 'Kribi', 'Sangmélima']),
    StateData(name: 'Adamawa', cities: ['Ngaoundéré', 'Meiganga', 'Tignère']),
    StateData(name: 'North', cities: ['Garoua', 'Guider', 'Pitoa']),
    StateData(name: 'Far North', cities: ['Maroua', 'Kousseri', 'Mokolo']),
  ]),
  CountryData(name: 'Senegal', states: [
    StateData(name: 'Dakar', cities: ['Dakar', 'Pikine', 'Guédiawaye']),
    StateData(name: 'Thiès', cities: ['Thiès', 'Mbour', 'Tivaouane']),
    StateData(name: 'Saint-Louis', cities: ['Saint-Louis', 'Richard-Toll', 'Dagana']),
    StateData(name: 'Diourbel', cities: ['Diourbel', 'Mbacké', 'Bambey']),
    StateData(name: 'Kaolack', cities: ['Kaolack', 'Nioro du Rip', 'Guinguinéo']),
    StateData(name: 'Ziguinchor', cities: ['Ziguinchor', 'Bignona', 'Oussouye']),
    StateData(name: 'Louga', cities: ['Louga', 'Linguère', 'Kébémer']),
    StateData(name: 'Fatick', cities: ['Fatick', 'Sokone', 'Foundiougne']),
    StateData(name: 'Tambacounda', cities: ['Tambacounda', 'Bakel', 'Kidira']),
    StateData(name: 'Kolda', cities: ['Kolda', 'Vélingara', 'Saraya']),
  ]),
  CountryData(name: 'United States', states: [
    StateData(name: 'California', cities: ['Los Angeles', 'San Francisco', 'San Diego', 'Sacramento']),
    StateData(name: 'New York', cities: ['New York City', 'Buffalo', 'Rochester', 'Albany']),
    StateData(name: 'Texas', cities: ['Houston', 'Dallas', 'Austin', 'San Antonio']),
    StateData(name: 'Florida', cities: ['Miami', 'Orlando', 'Tampa', 'Jacksonville']),
    StateData(name: 'Illinois', cities: ['Chicago', 'Springfield', 'Peoria']),
    StateData(name: 'Georgia', cities: ['Atlanta', 'Savannah', 'Augusta']),
    StateData(name: 'Maryland', cities: ['Baltimore', 'Annapolis', 'Silver Spring']),
    StateData(name: 'New Jersey', cities: ['Newark', 'Jersey City', 'Trenton']),
    StateData(name: 'Virginia', cities: ['Richmond', 'Virginia Beach', 'Arlington']),
    StateData(name: 'Massachusetts', cities: ['Boston', 'Cambridge', 'Worcester']),
  ]),
  CountryData(name: 'United Kingdom', states: [
    StateData(name: 'England', cities: ['London', 'Manchester', 'Birmingham', 'Liverpool', 'Leeds']),
    StateData(name: 'Scotland', cities: ['Edinburgh', 'Glasgow', 'Aberdeen', 'Dundee']),
    StateData(name: 'Wales', cities: ['Cardiff', 'Swansea', 'Newport']),
    StateData(name: 'Northern Ireland', cities: ['Belfast', 'Derry', 'Lisburn']),
  ]),
  CountryData(name: 'Canada', states: [
    StateData(name: 'Ontario', cities: ['Toronto', 'Ottawa', 'Mississauga', 'Hamilton']),
    StateData(name: 'Quebec', cities: ['Montreal', 'Quebec City', 'Laval', 'Gatineau']),
    StateData(name: 'British Columbia', cities: ['Vancouver', 'Victoria', 'Surrey', 'Burnaby']),
    StateData(name: 'Alberta', cities: ['Calgary', 'Edmonton', 'Red Deer']),
    StateData(name: 'Manitoba', cities: ['Winnipeg', 'Brandon', 'Steinbach']),
    StateData(name: 'Saskatchewan', cities: ['Saskatoon', 'Regina', 'Prince Albert']),
    StateData(name: 'Nova Scotia', cities: ['Halifax', 'Sydney', 'Dartmouth']),
  ]),
  CountryData(name: 'Other', states: [
    StateData(name: 'Other', cities: ['Other']),
  ]),
];

List<StateData> getStatesForCountry(String countryName) {
  final country = countries.where((c) => c.name == countryName).toList();
  return country.isNotEmpty ? country.first.states : [];
}

List<String> getCitiesForState(String countryName, String stateName) {
  final states = getStatesForCountry(countryName);
  final state = states.where((s) => s.name == stateName).toList();
  return state.isNotEmpty ? state.first.cities : [];
}

const List<String> referralSources = [
  'Google Search',
  'TikTok',
  'WhatsApp Status',
  'Xlounge-Extreme',
  'From a friend or Family member',
  'Facebook',
  'Myngul',
  'Twitter',
  'Instagram',
  'Youtube',
  'Telegram',
  'Other, please specify',
];
